### PR TITLE
pipeline(ticdc): add actor node

### PR DESCRIPTION
**Pull request by sdojjy**
### What problem does this PR solve?

Issue Number: ref #3881

this is a part of table pipeline actor model refactor #3881, 
this PR add actor node to process table event non-blocking

### Check List 

Tests 

 - Unit test
 - Integration test
 - 

### Release note 

```release-note
 `None`.
```

### CODE REVIEW

File: cdc/processor/pipeline/actor_node.go [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+package pipeline
+
+import (
+	"context"
+
+	"github.com/pingcap/errors"
+	"github.com/pingcap/tiflow/pkg/pipeline"
+)
+
+// ActorNode is an async data process node, it fetches and process table event non-blocking
```
Comment (asddongmen) [inline cdc/processor/pipeline/actor_node.go:23]:
```suggestion
// ActorNode is an async data process node, it fetches and handle table event non-blocking
```

File: cdc/processor/pipeline/actor_node.go [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+// if processing is blocked, the data will be cached and wait next run
+type ActorNode struct {
+	eventStash    *pipeline.Message
+	parentNode    AsyncDataHolder
```
Comment (asddongmen) [inline cdc/processor/pipeline/actor_node.go:27]: may be we can unified the name of an `event` / `data` / `message`?

Comment (sdojjy) [reply cdc/processor/pipeline/actor_node.go]: good sugestion

File: cdc/processor/pipeline/actor_node.go [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+
+// NewActorNode create a new ActorNode
+func NewActorNode(parentNode AsyncMessageHolder, messageProcessor AsyncMessageProcessor) *ActorNode {
+	return &ActorNode{
+		parentNode:       parentNode,
+		messageProcessor: messageProcessor,
+	}
+}
+
+// TryRun get message and process it util no message is available or message processing is blocked
```
Comment (asddongmen) [inline cdc/processor/pipeline/actor_node.go:39]:
```suggestion
// TryRun get message from parentNode and handle it util there is no more message to come
//  or message handling is blocking
```

File: cdc/processor/pipeline/actor_node.go [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+}
+
+// AsyncMessageProcessor is an interface to handle message non-blocking
+type AsyncMessageProcessor interface {
+	TryHandleDataMessage(ctx context.Context, msg pipeline.Message) (bool, error)
+}
+
+// AsyncMessageHolder is an interface to get message non-blocking
+type AsyncMessageHolder interface {
+	TryGetProcessedMessage() *pipeline.Message
```
Comment (asddongmen) [inline cdc/processor/pipeline/actor_node.go:71]: I think the  func name `TryGetProcessedMessage`  is verbose , may be we can just name it `TryGetDataMessage`.

Comment (sdojjy) [reply cdc/processor/pipeline/actor_node.go]: done

**Review by asddongmen — APPROVED** [review]

File: cdc/processor/pipeline/actor_node.go [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+
+// TryRun get message from parentNode and handle it util there is no more message to come
+//  or message handling is blocking
+// only one message will be cached
+func (n *ActorNode) TryRun(ctx context.Context) error {
+	for {
+		// batch?
+		if n.messageStash == nil {
+			n.messageStash = n.parentNode.TryGetDataMessage()
+		}
```
Comment (overvenus) [inline cdc/processor/pipeline/actor_node.go:47]: Will it be a busy loop?

Comment (sdojjy) [reply cdc/processor/pipeline/actor_node.go:47]: if can not get message or can not process message ,  will break the for loop

**Review by overvenus — APPROVED** [review]

**Pull request merged**